R/trackhub.R

```diff
@@ -0,0 +1,633 @@
+                        graphTypeDefault = "character",
+                        maxWindowToQuery = "character",
+                        negateValues = "character",
+                        smoothingWindow = "character",
+                        transformFunc = "character",
+                        windowingFunction = "character",
+                        yLineMark = "character",
+                        yLineOnOff = "character",
+                        gridDefault = "character",
+
+
+                        # hic settings
+                        showSnpWidth = "character",
+                        otherSpecies = "character",
+
+
+                        # vcfTabix settings
+                        minQual = "character",
+                        minFreq = "character",
+                        hapClusterEnabled = "character",
+                        hapClusterColorBy = "character",
+                        hapClusterTreeAngle = "character",
+                        hapClusterHeight = "character",
+                        applyMinQual = "character",
+
+
+                        ##Grouping tracks into sets and hierarchies
+
+                        # Supertrack
+                        superTrack = "character",
+                        parent = "character",
+
+                        # Composite Tracks
+                        compositeTrack = "character",
+                        allButtonPair = "character",
+                        centerLabelsDense = "character",
+                        dragAndDrop = "character",
+
+                        # Subgroups
+                        subGroup1 = "character",
+                        subGroup2 = "character",
+                        subGroup3 = "character",
+                        subGroup4 = "character",
+                        subGroup5 = "character",
+                        subGroup6 = "character",
+                        subGroup7 = "character",
+                        subGroup8 = "character",
+                        subGroup9 = "character",
+                        subGroups = "character",
+                        dimensions = "character",
+                        filterComposite = "character",
+                        dimensionAchecked = "character",
+                        dimensionBchecked = "character",
+                        sortOrder = "character",
+
+                        # Subgroups Views
+                        view = "character",
+                        viewUi = "character",
+                        configurable = "character",
+
+                        # multiWig
+                        container = "character",
+                        aggregate = "character",
+                        showSubtrackColorOnUi = "character",
+
+                        # Miscellaneous Deprecated Settings
+                        metadata = "character",
+                        noInherit = "character",
+                        useScore = "character"
```

Comment:
Are you sure all of these are of type "character"? For example, `useScore` might be logical.

Reply:
yes not all fields are up of "character" but the way we are creating `Track` object

[https://github.com/sanchit-saini/rtracklayer/blob/590d141a8213dfa86a11697ccd83a70a6a950f23/R/trackhub.R#L267](https://github.com/sanchit-saini/rtracklayer/blob/590d141a8213dfa86a11697ccd83a70a6a950f23/R/trackhub.R#L267)

so in order to make slots up of different type, first we need to remove `'` from non-character value and parse them to convert the value to the right data type.

```R
useScore 1 # file input
useScore = TRUE 
``` 

or maybe we can directly pass all arguments with `'` later on in Track definition handle task of removing and parsing of value.
[https://github.com/sanchit-saini/rtracklayer/blob/590d141a8213dfa86a11697ccd83a70a6a950f23/R/trackhub.R#L619](https://github.com/sanchit-saini/rtracklayer/blob/590d141a8213dfa86a11697ccd83a70a6a950f23/R/trackhub.R#L619)


And also we need to revert these converted values to their original form to write back to the file so maybe we could build generalized methods for parsing and de-parsing similar kinds of inputs (e.g on and off).

does this approach seem reasonable?

Reply:
Yes, that's a great idea. Separate the on disk representation from the object model through marshaling and unmarshaling transformations.